src/mpi/request/mpir_request.c

```diff
@@ -127,12 +123,7 @@ int MPIR_Request_completion_processing(MPIR_Request * request_ptr, MPI_Status *
 
         case MPIR_REQUEST_KIND__GREQUEST:
             {
-                int rc;
-
-                rc = MPIR_Grequest_query(request_ptr);
-                if (mpi_errno == MPI_SUCCESS) {
-                    mpi_errno = rc;
-                }
+                mpi_errno = MPIR_Grequest_query(request_ptr);
                 MPIR_Request_extract_status(request_ptr, status);
                 rc = MPIR_Grequest_free(request_ptr);
```

Comment:
@danghvu sorry for the long delay in reviewing. My only comment is here. I think we should do this same simplification for `MPIR_Grequest_free`. It and `MPIR_Grequest_query` return MPI error codes, so there is no need for the intermediate `rc` variable.

Reply:
OK I see what you mean now. Sorry for the long delay again. I'm going to rebase/retest and see if hopefully this is ready for merge.